app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulation.java

```diff
@@ -49,6 +51,16 @@ public DynamoSubpopulation() {
     }
     
     @JsonIgnore
+    @DynamoDBIgnore
+    public String getKey() {
+        return "subpopulation:"+guid;
+    }
+    
+    @Override
+    public void setKey(String key) {
+    }
```

Comment:
I think it would be cleaner to have Subpopulation no longer implement the Criteria interface. This ensures that we stop passing in Subpopulation when we actually want a Criteria, so we can cleanly finish the refactor. This also means we don't have to do silly things like implement getKey() and setKey(), when we really don't need them.